### PR TITLE
Update vm_service dependency.

### PR DESCRIPTION
**Pull request by jacob314**
To make rolling in the future easier, removed the dependency on package:test from package:devtools_app. This avoids problems in cases where there isn't a version of package:test that is compatible with all of the other packages required by devtools_app. All of the tests from devtools_app can be run with flutter_test anyway so there isn't much of a loss of functionality by switching to flutter_test.

### CODE REVIEW

File: packages/devtools_app/pubspec.yaml [hunk 1]
```diff
@@ -52,9 +52,9 @@ dependencies:
   string_scanner: ^1.1.0
   url_launcher: ^5.0.0
   url_launcher_web: ^0.1.1+6
-  vm_service: ^6.2.0
+  vm_service: ^7.1.0
```
Comment (kenzieschmoll) [inline packages/devtools_app/pubspec.yaml:55]: don't know if it matters, but the dds vm_service dep is ^7.0.0

Comment (jacob314) [reply packages/devtools_app/pubspec.yaml:55]: So ^7.0.0 is the same as '>=7.0.0 <8.0.0'

So the versions are compatible. As our dependency on vm_service is fragile as all changes are breaking due to implementing a class that isn't expected to be an interface, I like making our dependency a bit on the stricter side. `7.1.0` is the latest published version.

**Comment by jacob314** [comment]
This change got much bigger than I'd like due to version match pain. I stripped the dep on `package:test` to avoid version mismatch pain. I don't see any reason why we should be sometimes using package:test instead of package:flutter_test. Given how tightly bound our versions are to packages that are also package:test and package:flutter_test deps I think this is a reasonable tradeoff.

File: packages/devtools_app/pubspec.yaml [hunk 2]
```diff
@@ -130,3 +129,5 @@ dependency_overrides:
     path: ../devtools_shared #OVERRIDE_FOR_DEVELOPMENT
   devtools_testing: #OVERRIDE_FOR_DEVELOPMENT
     path: ../devtools_testing #OVERRIDE_FOR_DEVELOPMENT
+  devtools:  #OVERRIDE_FOR_DEVELOPMENT
+    path: ../devtools  #OVERRIDE_FOR_DEVELOPMENT
```
Comment (kenzieschmoll) [inline packages/devtools_app/pubspec.yaml:133]: why is this necessary if the dev dependency on devtools already depends on devtools from path? is devtools a transitive dep somewhere?

**Review by kenzieschmoll — COMMENTED** [review]
looks like devtools_app, devtools_testing, and devtools need their pubspec versions to be bumped to 2.3.1 also, as well dependency versions for those packages

File: packages/devtools_app/pubspec.yaml (hunk 2 above)
Comment (jacob314) [reply packages/devtools_app/pubspec.yaml:133]: I've now tweaked the other dependency to not be a path dependency for clarity. Using dependency overrides makes the version solving a little less frustrating.

**Comment by jacob314** [comment]
> looks like devtools_app, devtools_testing, and devtools need their pubspec versions to be bumped to 2.3.1 also, as well dependency versions for those packages. Might as well prep for the release here too since some of the package versions are changing.
Actually need to bump to 2.3.2. I'll send that a separate CL as I already have that update and the changelog tweak ready to go in a separate CL.

**Review by kenzieschmoll — APPROVED** [review]
